app/src/main/kotlin/org/koitharu/kotatsu/core/os/AppValidator.kt

```diff
@@ -1,46 +1,23 @@
 package org.koitharu.kotatsu.core.os
 
-import android.annotation.SuppressLint
 import android.content.Context
 import android.content.pm.PackageManager
+import androidx.core.content.pm.PackageInfoCompat
 import dagger.hilt.android.qualifiers.ApplicationContext
-import org.koitharu.kotatsu.core.util.ext.printStackTraceDebug
-import org.koitharu.kotatsu.parsers.util.byte2HexFormatted
-import java.io.ByteArrayInputStream
-import java.io.InputStream
-import java.security.MessageDigest
-import java.security.cert.CertificateFactory
-import java.security.cert.X509Certificate
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class AppValidator @Inject constructor(
 	@ApplicationContext private val context: Context,
 ) {
-
+	@Suppress("NewApi")
```

Comment:
Why is this suppression needed?

Reply:
It's for the PackageManager constant used in the map.

Reply:
Does it mean it will not work on older api?

Reply:
It will, the PackageManager value used is a constant that gets inlined during compilation.